### PR TITLE
Dataloader

### PR DESCRIPTION
**Pull request by kks32**
**Describe the PR**
Supports both npz and hdf5 data format

**Related Issues/PRs**
#82 

**Additional Context**
Will remove data_loader.py once we have merged multinode training to train.py

### CODE REVIEW

File: gns/train.py [hunk 1]
```diff
@@ -750,45 +799,30 @@ def validation(simulator, example, n_features, cfg, rank, device_id):
...
-            else:
-                world_size = cfg.hardware.n_gpus
-
-            # Print the status of GPU usage
-            print(f"Using {world_size}/{available_gpus} GPUs")
-
-            # Spawn training to GPUs
-            distribute.spawn_train(train, cfg, world_size, device)
+            torch.multiprocessing.set_start_method("spawn")
+            verbose, world_size = distribute.setup(local_rank)
```
Comment (yjchoi1) [inline gns/train.py:817]: Running it on single node shows the following error:

```
Error executing job with overrides: []
Traceback (most recent call last):
  File "/work2/08264/baagee/frontera/gns-main/gns/train.py", line 817, in main
    verbose, world_size = distribute.setup(local_rank)
UnboundLocalError: local variable 'local_rank' referenced before assignment

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.

```

**Comment by yjchoi1** [comment]
Now, it works with my custom data on a single node and multi-nodes. I haven't checked with `.h5` though.

File: gns/train.py
Comment (yjchoi1) [inline gns/train.py:1]: In the resume mode, `simulator.module.load()` seems to be changed to `simulator.load()` although the previous version of parallel GNS used to work with `simulator.load()`.
```
Error executing job with overrides: []
Traceback (most recent call last):
  File "/work2/08264/baagee/frontera/gns-main/gns/train.py", line 843, in main
    train(local_rank, cfg, world_size, device, verbose, use_dist)
  File "/work2/08264/baagee/frontera/gns-main/gns/train.py", line 475, in train
    simulator.module.load(cfg.model.path + cfg.model.file)
  File "/work2/08264/baagee/frontera/venvs/venv-frontera-gpu/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1207, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'LearnedSimulator' object has no attribute 'module'
```

**Comment by yjchoi1** [comment]
Also, there is a minor error about `tqdm` count. 

When `resume`, `tqdm` does not show the correct `nstep`. For example, given that the model is saved at 100/500000 step at epoch=0, when we resume from `model-100.pt`, `tqdm` count starts from 0/500000, although the code correctly loaded the model at 100 step.

File: gns/train.py
Comment (kks32) [reply gns/train.py:1]: @skye-glitch : Could you address this please.

Comment (skye-glitch) [reply gns/train.py:1]: Please check the updated code.

**Comment by skye-glitch** [comment]
> Also, there is a minor error about `tqdm` count.
> 
> When `resume`, `tqdm` does not show the correct `nstep`. For example, given that the model is saved at 100/500000 step at epoch=0, when we resume from `model-100.pt`, `tqdm` count starts from 0/500000 rather than starting from 100/500000, although the code correctly loaded the model at 100 step.

Should have been fixed

**Comment by yjchoi1** [comment]
Another minor issue is that gns/args.py still has `n-gpus` arg although we don't use them.

**Comment by skye-glitch** [comment]
> Another minor issue is that `gns/args.py` still has `n_gpus` arg although we don't use them.

Fixed. Thanks！

**Comment by kks32** [comment]
@yjchoi1 Could you check to see if everything is good to merge?

**Comment by yjchoi1** [comment]
After addressing above comment about single node issue, everything seems good to merge.

**Pull request merged**